corehq/form_processor/abstract_models.py

```diff
@@ -37,6 +37,10 @@ def set_submission_properties(self, submission_post):
     def to_json(self):
         raise NotImplementedError()
 
+    @property
+    def is_normal(self):
```

Comment:
probably a bad idea but you could add some `state` variable or something to the class and then use it to keep track of all the attributes you want to add